Composer/packages/adaptive-form/src/utils/arrayUtils.ts

```diff
@@ -78,6 +79,19 @@ export function useArrayItems<ItemType = unknown>(
 ): ArrayItemState<ItemType> {
   const [cache, setCache] = useState(generateArrayItems(items));
 
+  useEffect(() => {
+    const newCache = generateArrayItems(items);
+
+    if (
+      !isEqual(
+        cache.map(({ value }) => value),
```

Comment:
This could yield undefined === undefined right?

Reply:
I don't think so. If a value is `undefined`, the Adaptive Form removes it from the array.